hbase-server/src/main/java/org/apache/hadoop/hbase/mob/DefaultMobStoreCompactor.java

```diff
@@ -362,11 +508,375 @@ protected boolean performCompaction(FileDetails fd, InternalScanner scanner, Cel
         abortWriter(mobFileWriter);
       }
     }
+    // Commit or abort generational writers
+    if (mobWriters != null) {
+      for (StoreFileWriter w: mobWriters.getOutputWriters()) {
+        Long mobs = mobWriters.getMobCountForOutputWriter(w);
+        if (mobs != null && mobs > 0) {
+          mobRefSet.get().add(w.getPath().getName());
+          w.appendMetadata(fd.maxSeqId, major, mobs);
+          w.close();
+          mobStore.commitFile(w.getPath(), path);
+        } else {
+          abortWriter(w);
```

Comment:
log at debug that we're aborting this one because there are no cells.

Reply:
Fixed